std/encoding.d

```diff
@@ -2233,7 +2101,7 @@ body
     }
     else static if(is(Src==AsciiChar))
     {
-        transcode!(char,Dst)(cast(string)s,r);
+        transcodeFromAscii(s, r);
```

Comment:
> Please instead do:

Done.

> I thought Andrei was strongly against this trusted lambda pattern.

Trusted lambda pattern is this:

    void foo() @safe {
      (() @trusted { doSomethingEvil(); })();
    }

This is the trusted template overload to work around a compiler bug that doesn't see const enum array -> const value-type array as @safe pattern. See https://issues.dlang.org/show_bug.cgi?id=15762

> It looks like all possible instantiations of transcode are memory safe. In that case @trusted should be on transcode in its entirety.

You mean @safe, no? @trusted would make the whole function opt out of static checking, which is not ideal. But again, that compiler bug prevents it -- which is why I have the `// @@@BUG@@@ 15762` comment there.